noir-projects/aztec-nr/aztec/src/encrypted_logs/body.nr

```diff
@@ -0,0 +1,147 @@
+use dep::protocol_types::{grumpkin_private_key::GrumpkinPrivateKey, grumpkin_point::GrumpkinPoint};
+
+use crate::oracle::encryption::aes128_encrypt;
+use crate::keys::point_to_symmetric_key::point_to_symmetric_key;
+
+struct EncryptedLogBody<Note> {
+    storage_slot: Field,
+    note_type_id: Field,
+    note: Note,
+}
+
+impl<Note> EncryptedLogBody<Note> {
+    pub fn new<N>(
+        storage_slot: Field,
+        note_type_id: Field,
+        note: Note
+    ) -> Self where Note: NoteInterface<N> {
+        Self { storage_slot, note_type_id, note }
+    }
+
+    pub fn compute_ciphertext<N, M>(
+        self,
+        secret: GrumpkinPrivateKey,
+        point: GrumpkinPoint
+    ) -> [u8; M] where Note: NoteInterface<N> {
+        // We need 32 bytes for every field in the note, and then we have 2 extra fields (storage_slot and note_type_id)
+        let serialized_note: [Field; N] = Note::serialize_content(self.note);
+
+        // Work around not being able to use N directly beyond the size of the array above.
+        let N_ = serialized_note.len();
+
+        assert(N_ * 32 + 64 == M, "Invalid size of encrypted log body");
+
+        let mut buffer: [u8; M] = [0; M];
+
+        let storage_slot_bytes = self.storage_slot.to_be_bytes(32);
+        let note_type_id_bytes = self.note_type_id.to_be_bytes(32);
+        for i in 0..32 {
+            buffer[i] = storage_slot_bytes[i];
+            buffer[32 + i] = note_type_id_bytes[i];
+        }
+
+        for i in 0..N_ {
+            let bytes = serialized_note[i].to_be_bytes(32);
+            for j in 0..32 {
+                buffer[64 + i * 32 + j] = bytes[j];
+            }
+        }
+
+        let full_key = point_to_symmetric_key(secret, point);
+        let mut sym_key = [0; 16];
+        let mut iv = [0; 16];
+
+        for i in 0..16 {
+            sym_key[i] = full_key[i];
+            iv[i] = full_key[i + 16];
+        }
+
+        aes128_encrypt(buffer, iv, sym_key)
+    }
+}
+
+/*
+// Test is semi broken, needs to be fixed along with #6172  
+mod test {
+    use crate::encrypted_logs::body::EncryptedLogBody;
+    use dep::protocol_types::{address::AztecAddress, traits::Empty, constants::GENERATOR_INDEX__NOTE_NULLIFIER};
+
+    use crate::{
+        note::{note_header::NoteHeader, note_interface::NoteInterface, utils::compute_note_hash_for_consumption},
+        oracle::{unsafe_rand::unsafe_rand, nullifier_key::get_app_nullifier_secret_key, get_public_key::get_public_key},
+        context::PrivateContext, hash::poseidon2_hash
+    };
+
+    use dep::protocol_types::{address::AztecAddress, grumpkin_private_key::GrumpkinPrivateKey, grumpkin_point::GrumpkinPoint};
+
+    struct AddressNote {
+        address: AztecAddress,
+        owner: AztecAddress,
+        randomness: Field,
+        header: NoteHeader,
+    }
+
+    global BIB_BOB_ADDRESS_NOTE_LEN: Field = 3;
+
+    impl NoteInterface<BIB_BOB_ADDRESS_N3OTE_LEN> for AddressNote {
```

Comment:
```suggestion
    impl NoteInterface<BIB_BOB_ADDRESS_NOTE_LEN> for AddressNote {
```

Reply:
💀 will fix it as part of #6172 where the test is to also be made to work